Fix test_bibtex unit test

**Describe the bug**
The `test_bibtex` unit test with failing, with a cryptic traceback


**Screenshots**
```
======================================================================
FAIL: test_bibtex (test_module.TestScholarly)
Test that we get the BiBTeX entry correctly
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/scholarly/scholarly/test_module.py", line 233, in test_bibtex
    self.assertEqual(result, expected_result.replace("\n        ", "\n"))
AssertionError: '@inp[134 chars]ses.  However, the well-known algorithms suffe[4[71](https://github.com/scholarly-python-package/scholarly/runs/5045402146?check_suite_focus=true#step:8:71) chars]\n\n' != '@inp[134 chars]ses. However, the application to large spatial[457 chars]\n\n'
Diff is 960 characters long. Set self.maxDiff to None to see it.
```

**Desktop (please complete the following information):**
Github Action

**Do you plan on contributing?**
- [ ] Yes, I will create a Pull Request with the bugfix.

--- a/.github/CONTRIBUTING.md
+++ b/.github/CONTRIBUTING.md
@@ -16,12 +16,13 @@ Additionally, if you are interesting in contributing to the codebase, submit a p
 
 ## How to contribute
 
-1. Create a fork of `scholarly-python-package/scholarly` repository.
-2. If you add a new feature, try to include tests in already existing test cases, or create a new test case if that is not possible.
-3. Make sure the unit tests pass before raising a PR. For all the unit tests to pass, you typically need to setup a premium proxy service such as `ScraperAPI` or `Luminati` (`Bright Data`). If you do not have an account, you may try to use `FreeProxy`. Without a proxy, 6 out of 17 test cases will be skipped.
-4. Check that the documentatation is consistent with the code. Check that the documentation builds successfully.
-5. Submit a PR, with `develop` as your base branch.
-6. After an initial code review by the maintainers, the unit tests will be run with the `ScraperAPI` key stored in the Github repository. Passing all tests cases is necessary before merging your PR.
+1. Create a fork of `scholarly-python-package/scholarly` repository. Make sure that "Copy the main branch only" is **not** checked off.
+2. After cloning your fork and checking out into the develop branch, run `python setup.py --help-commands` for more info on how to install dependencies and build. You may need to run it with `sudo`.
+3. If you add a new feature, try to include tests in already existing test cases, or create a new test case if that is not possible. For a comprehensive output, run `python -m unittest -v test_module.py`
+4. Make sure the unit tests pass before raising a PR. For all the unit tests to pass, you typically need to setup a premium proxy service such as `ScraperAPI` or `Luminati` (`Bright Data`). By default, `python setup.py install` will get `FreeProxy`. Without a proxy, 6 out of 17 test cases will be skipped.
+5. Check that the documentatation is consistent with the code. Check that the documentation builds successfully.
+6. Submit a PR, with `develop` as your base branch.
+7. After an initial code review by the maintainers, the unit tests will be run with the `ScraperAPI` key stored in the Github repository. Passing all tests cases is necessary before merging your PR.
 
 
 ## Build Docs

--- a/test_module.py
+++ b/test_module.py
@@ -690,7 +690,7 @@ class TestScholarlyWithProxy(unittest.TestCase):
         Test that we get the BiBTeX entry correctly
         """
 
-        with open("testdata/bibtex.txt", "r") as f:
+        with open("testdata/test_bibtex_result.txt", "r") as f:
             expected_result = "".join(f.readlines())
 
         pub = scholarly.search_single_pub("A distribution-based clustering algorithm for mining in large "

--- a/test_module.py
+++ b/test_module.py
@@ -78,6 +78,20 @@ class TestTorInternal(unittest.TestCase):
         authors = [a for a in scholarly.search_author(query)]
         self.assertGreaterEqual(len(authors), 1)
 
+class TestFreeProxy(unittest.TestCase):
+    luminati = os.getenv("USERNAME") and os.getenv("PASSWORD") and os.getenv("PORT")
+    scraperAPI = os.getenv('SCRAPER_API_KEY')
+    skipIf = (luminati is not None) or (scraperAPI is not None)
+
+    @unittest.skipIf(skipIf, reason="Other proxy is being used")
+    def test_freeproxy(self):
+        """
+        Test that we can set up FreeProxy successfully
+        """
+        proxy_generator = ProxyGenerator()
+        success = proxy_generator.FreeProxies()
+        self.assertTrue(success)
+        self.assertEqual(proxy_generator.proxy_mode, "FREE_PROXIES")
 
 class TestScholarly(unittest.TestCase):
 

